src/core/atomic.d

```diff
@@ -157,6 +157,38 @@ version( CoreDdoc )
 }
 else version( AsmX86_32 )
 {
+    // Uses specialized asm for fast fetch and add operations
+    private HeadUnshared!(T) atomicFetchAdd(T)( ref shared T val, size_t mod ) pure nothrow @nogc
```

Comment:
Even better, we keep this as `atomicFetchAdd`, and do the post adding in `atomicOp`.
This is because C11 and C++ only standardize `[atomic_fetch_add](http://en.cppreference.com/w/cpp/atomic/atomic_fetch_add)`.
So this function should simply `return cast(T)tmp;` and atomicOp should instead `return cast(T)(atomicFetchAdd!(T)(val, mod) + mod);`.